homeassistant/components/zha/config_flow.py

```diff
@@ -92,6 +112,74 @@ async def async_step_pick_radio(self, user_input=None):
             data_schema=vol.Schema(schema),
         )
 
+    async def async_step_usb(self, discovery_info: DiscoveryInfoType):
+        """Handle usb discovery."""
+        vid = discovery_info["vid"]
+        pid = discovery_info["pid"]
+        serial_number = discovery_info["serial_number"]
+        device = discovery_info["device"]
+        manufacturer = discovery_info["manufacturer"]
+        description = discovery_info["description"]
+        await self.async_set_unique_id(
+            f"{vid}:{pid}_{serial_number}_{manufacturer}_{description}"
```

Comment:
Wouldn't just serial number be enough ? What if they plug it into a different port?

Reply:
Some of the no name devices don't expose a serial number. I used all of the data since that's what device by id on the file system seems to do